document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresSelectTypeExpressionVisitor.java

```diff
@@ -60,13 +64,25 @@ public static String getSelections(final List<SelectionSpec> selectionSpecs) {
     PostgresIdentifierExpressionVisitor identifierExpressionVisitor =
         new PostgresIdentifierExpressionVisitor();
 
+    Map<String, String> pgSelections = postgresQueryParser.getPgSelections();
+
     String childList =
         selectionSpecs.stream()
             .map(
-                selectionSpec ->
-                    new PgSelection(
-                        selectionSpec.getExpression().accept(selectTypeExpressionVisitor),
-                        getAlias(selectionSpec, identifierExpressionVisitor)))
+                selectionSpec -> {
+                  PgSelection pgSelection = new PgSelection();
+                  pgSelection.fieldName =
+                      selectionSpec.getExpression().accept(selectTypeExpressionVisitor);
+                  if (!StringUtils.isEmpty(selectionSpec.getAlias())) {
+                    pgSelection.alias = selectionSpec.getAlias();
+                    pgSelections.put(pgSelection.alias, pgSelection.fieldName);
+                  } else {
+                    pgSelection.alias =
+                        PostgresUtils.encodeAliasForNestedField(
+                            selectionSpec.getExpression().accept(identifierExpressionVisitor));
+                  }
+                  return pgSelection;
```

Comment:
Too many things happening inside a lambda expression used inside `map`. Can we break it into a methods of its own?

Reply:
Done. Refactored a bit.